build_runner/lib/src/logging/std_io_logging.dart

```diff
@@ -57,34 +57,11 @@ StringBuffer colorLog(LogRecord record, {bool verbose}) {
 void _stdIOLogListener(LogRecord record, {bool verbose}) =>
     stdout.write(colorLog(record, verbose: verbose));
 
-/// Filter out the Logger names known to come from `build_runner` and splits the
-/// header for levels >= WARNING.
-String _loggerName(LogRecord record, bool verbose) {
-  var knownNames = const [
-    'ApplyBuilders',
-    'Bootstrap',
-    'Build',
-    'BuildConfigOverrides',
-    'BuildDefinition',
-    'BuildOptions',
-    'BuildScriptUpdates',
-    'CreateOutputDir',
-    'Entrypoint',
-    'Heartbeat',
-    'IOEnvironment',
-    'Serve',
-    'Watch',
-    'build_runner',
-    // commands
-    'build',
-    'clean',
-    'doctor',
-    'serve',
-    'test',
-    'watch',
-  ];
+/// Filter out the Logger names which aren't coming from specific builders and
+/// splits the header for levels >= WARNING.
+String _recordHeader(LogRecord record, bool verbose) {
   var maybeSplit = record.level >= Level.WARNING ? '\n' : '';
-  return verbose || !knownNames.contains(record.loggerName)
+  return verbose || record.loggerName.contains(' on ')
```

Comment:
Not a big fan of the pattern matching here... if we instead removed all the custom loggers and used a single common logger we could check for that one specifically and it would be a fair bit better I think?

Reply:
Oh although then we wouldn't get the names in verbose mode...

Reply:
Another option could be creating a custom LogRecord type? That would be safer but also has other disadvantages, although that class hasn't changed in ages so it is unlikely we would run into issues.

Reply:
I'd be in favor of dropping the names in verbose mode and simplifying down to a single common logger, but they can be handy sometimes so I can see an argument for keeping them.

I'm not excited by the idea of a custom `LogRecord` implementation, though I think all the cases for logs that we _do_ want to print the name go through [`BuildForInputLogger`](https://github.com/dart-lang/build/blob/83ea53f8dcfef04e17784698656c376f7e3e58c4/build_runner_core/lib/src/logging/build_for_input_logger.dart) which might give us some leverage to handle it...

Reply:
https://github.com/dart-lang/webdev/pull/305 added a dependency on the only logger names we want printed containing a space. I'll use the same criteria here.

In the long term we'll likely want to prefix all the non-shown logger names with `build.`

Reply:
> In the long term we'll likely want to prefix all the non-shown logger names with `build.`

Although now that I think about it this would break for webdev haha